leejinwoo/src/components/Seats.tsx

```diff
@@ -0,0 +1,398 @@
+    const [movie, setMovie] = useState<number>(1);
+    // 영화 이름
+    const [movieName, setMovieName] = useState<string>('올빼미');
+    const [result, setResult] = useState<number>(14000);
+    // 라우터쓸거임ㅇㅇ
+    const router = useRouter();
+
+    //영화가 바뀌면 영화 이름을 바꾸게함
+    const onSelectChange = (e:ChangeEvent<HTMLSelectElement>) =>{
+        setMovie(Number(e.target.value));
+
+        if(e.target.value == '1'){
+            setMovieName(String('올빼미'));
+        }
+        if(e.target.value == '2'){
+            setMovieName(String('앙상블 스타즈'));
+        }
+        if(e.target.value == '3'){
+            setMovieName(String('오늘 밤, 세계에서 이 사랑이 사라진다 해도'));
+        }
+        if(e.target.value == '4'){
+            setMovieName(String('탄생'));
+        }
+        if(e.target.value == '5'){
+            setMovieName(String('그래비티'));
+        }
+        if(e.target.value == '6'){
+            setMovieName(String('더 메뉴'));
+        }
+        if(e.target.value == '7'){
+            setMovieName(String('프레이 포 더 데블'));
+        }
+        if(e.target.value == '8'){
+            setMovieName(String('매드맥스-분노의도로'));
+        }
+        if(e.target.value == '9'){
+            setMovieName(String('압꾸정'));
+        }
+        if(e.target.value == '10'){
+            setMovieName(String('크리스마스 캐럴'));
+        }
+    };
+
+    //조조할인은 못참지
+    const movePriceChange = (e: ChangeEvent<HTMLSelectElement>) => {
+        setPrice(Number(e.target.value));
+    };
+
+    //사람이 선택될 때 마다 가격이 변동 됨
+    useEffect(() => {
+        setResult(price * people)
+    },[price, people]);//[의존성 배열]
+
+    //선택 안하면 초기화
+    const onReset = () => {
+        setPeople(0);
+    };
+
+    //좌석 선택
+    const choiseChange = (e: React.MouseEvent<HTMLElement>) => {
+        // console.log(e.target);
+        // console.log(e.currentTarget.classList.contains('occu'));
+        //선택한 좌석이 이미 선택한 좌석이라면
+        const SeatStyle = e.currentTarget;
+        if(e.currentTarget.classList.contains('occupied')){
+            //알림창 생성"이미 선택된 좌석입니다"
+            alert("이미 선택된 좌석입니다")
+            return;
+            
+        }
+        //선택된 좌석을 다시 누르면
+        else if (e.currentTarget.classList.contains('selected')){
+            //사람 -1
+            setPeople(people -1);
+            SeatStyle.style.backgroundColor = '#fff';
+            SeatStyle.classList.remove('selected');
+
+        }
+        //선택한 좌석이 빈 좌석이라면
+        else{
+            setPeople(people +1);
+            SeatStyle.classList.add('selected');
+            SeatStyle.style.backgroundColor = '#0000ff';
+        }
+
+
+
+    };
+    //
+
+    //클릭되면 Purchase로 넘어가는 라우터
+    const onClickPurchase = () => {
+        if (people <= 0) {
+            alert('좌석을 선택하세요');
+            onReset();
+            setPrice(14000);
+            return;
+        }
+        else router.push({
+            pathname: '/Purchase',
+            // 해당 값들을 purchase로 넘겨라
+            query: {movie: movieName, people: people, price: result },
+        });
```

Comment:
```suggestion
        router.push({
            pathname: '/Purchase',
            // 해당 값들을 purchase로 넘겨라
            query: {movie: movieName, people: people, price: result },
        });
```
여기서는 위에  return을 하고 있기 떄문에 `else`는 사실 없어도 돼요

이게 위에서 말한 `early return` 방법 !

Reply:
👍